types/react/index.d.ts

```diff
@@ -204,7 +204,7 @@ declare namespace React {
         props?: ClassAttributes<T> & P,
         ...children: ReactNode[]): ReactSVGElement;
     function createElement<P extends DOMAttributes<T>, T extends Element>(
-        type: string,
+        type: ReactType,
         props?: ClassAttributes<T> & P,
         ...children: ReactNode[]): DOMElement<P, T>;
```

Comment:
This return type is no longer correct after your change. Consider adding another override as a catch-all, which takes `ReactType` and returns `ReactElement`.